apps/api/src/modules/providers/providers.service.ts

```diff
@@ -1,20 +1,51 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, Logger } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { Repository } from 'typeorm';
 import { Provider } from './entities/provider.entity';
 import { Status } from 'src/common/constants/database';
+import { CreateProviderInput } from './dto/create-provider.input';
+import { UsersService } from '../users/users.service';
+import { ApplicationsService } from '../applications/applications.service';
+import { QueryOptionsDto } from 'src/common/graphql/dtos/query-options.dto';
+import { ProviderResponse } from './dto/provider-response.dto';
+import { CoreService } from 'src/common/graphql/services/core.service';
 
 @Injectable()
-export class ProvidersService {
+export class ProvidersService extends CoreService<Provider> {
+  protected readonly logger = new Logger(ProvidersService.name);
   constructor(
     @InjectRepository(Provider)
     private readonly providerRepository: Repository<Provider>,
-  ) {}
+    private readonly applicationsService: ApplicationsService,
+    private readonly usersService: UsersService,
+  ) {
+    super(providerRepository);
+  }
 
   async getById(providerId: number): Promise<Provider | undefined> {
     return this.providerRepository.findOne({ where: { providerId, status: Status.ACTIVE } });
   }
 
+  async createProvider(
+    providerInput: CreateProviderInput,
+    authorizationHeader: Request,
+  ): Promise<Provider> {
+    const isAdmin = await this.applicationsService.checkAdminUser(authorizationHeader);
+
+    if (!isAdmin) {
+      throw new Error('Access Denied. Not an ADMIN.');
+    }
+
+    const userExists = await this.usersService.findByUserId(providerInput.userId);
+
+    if (!userExists) {
+      throw new Error('This user does not exist.');
+    }
+
+    const provider = this.providerRepository.create(providerInput);
```

Comment:
Before doing this, we also need to check if the provided applicationId and even channelType exist or not, else the foreign key checks will fail.
Can you verify this by trying to create a new provider with non-existent applicationId and/or channelType?

Reply:
Updated https://github.com/OsmosysSoftware/osmo-x/pull/257/commits/3af779fc6e3e47ae364c0e283f0c528b687cf123